src/Parallel/MemoryMonitor/MemoryMonitor.hpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "Parallel/Actions/SetupDataBox.hpp"
+#include "Parallel/Algorithms/AlgorithmSingletonDeclarations.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Parallel/Local.hpp"
+#include "Parallel/MemoryMonitor/Tags.hpp"
+#include "Parallel/ParallelComponentHelpers.hpp"
+#include "ParallelAlgorithms/Initialization/Actions/AddSimpleTags.hpp"
+#include "ParallelAlgorithms/Initialization/Actions/RemoveOptionsAndTerminatePhase.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace mem_monitor::detail {
+struct InitializeMutator {
+  using return_tags = tmpl::list<mem_monitor::Tags::MemoryHolder>;
+  using argument_tags = tmpl::list<>;
+
+  using tag_type = typename mem_monitor::Tags::MemoryHolder::type;
+
+  static void apply(const gsl::not_null<tag_type*> /*holder*/) {}
+};
+}  // namespace mem_monitor::detail
+
```

Comment:
can you document the `mem_monitor` namespace in this file? Not the `detail` one above, that way it'll show up in the dox, as well as everything inside of it. I would argue for putting the component in that namespace too. No reason all components have to be in the global namespace. Then I'd put the `mem_monitor` in the parallel doxygen group, but not the component itself